Applications/Spire/Source/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.cpp

```diff
@@ -0,0 +1,50 @@
+#include "Spire/OrderImbalanceIndicator/LocalOrderImbalanceIndicatorModel.hpp"
+
+using namespace boost::icl;
+using namespace Nexus;
+using namespace Spire;
+
+void LocalOrderImbalanceIndicatorModel::publish(
+    const Nexus::OrderImbalance& imbalance) {
+  m_publish_queue.push(imbalance);
+  if(m_publish_queue.size() > 1) {
+    return;
+  }
+  while(!m_publish_queue.empty()) {
+    const auto& current_imbalance = m_publish_queue.front();
+    if(!m_imbalances.insert(
+        {current_imbalance.m_security, current_imbalance}).second) {
+      if(auto& previous_imbalance = m_imbalances.at(imbalance.m_security);
+          previous_imbalance.m_timestamp < imbalance.m_timestamp) {
+        previous_imbalance = current_imbalance;
+      }
+    }
+    for(auto& subscription : m_subscriptions) {
+      if(contains(subscription.m_interval, current_imbalance.m_timestamp)) {
+        subscription.m_signal(current_imbalance);
+      }
+    }
+    m_publish_queue.pop();
+  }
+  std::erase_if(m_subscriptions,
```

Comment:
Do any erasing as part of the loop on line 22. As you iterate over ```m_subscriptions``` on line 22, you check if the signal is empty and remove it if it is.